### ISSUE TITLE
Reduce runtime node dependencies

### ISSUE THREAD

**Issue by felixdoerre** [mastodon/mastodon#21215]
### Pitch

We should split-up the `package.json` file, to have a separate set of dependencies for runtime that the ones required for building the static assets.

Concrete required steps would be:
- create a `package.json` that declares the runtime-dependencies 
  A possible first version for that would be:
```
{
  "dependencies": {
    "dotenv": "^16.0.3",
    "express": "^4.18.2",
    "http": "^0.0.1-security",
    "jsdom": "^20.0.1",
    "npmlog": "6.0.2",
    "pg": "^8.8.0",
    "redis": "4.0.6",
    "throng": "^5.0.0",
    "uuid": "^8.3.1",
    "ws": "^8.11.0"
  }
}
```
- use that `package.json` to install node-modules for runtime instead of copying the whole environment over from the build container.

### Motivation

Currently the node environment installed to `node_modules` is used in two contexts:
- for building the frontend pack-files
- for running the `streaming`-service

on my system the size of the current `node_modules` folder is `377M`. Installing only the dependencies required by `streaming` (stated in the first few lines of that file), I end up with a far smaller `node_modules` folder: `19M`

Getting the image size down allows for faster updates/reduced disk usage. Stripping down the runtime image is also security best-practice to only install things in production that are actually needed.

**Comment by ClearlyClaire** [comment]
The existing pull request #21143 aims to address this

**Comment by ThisIsMissEm** [comment]
We've work to do this post-4.2.0, which mostly requires updating the pull request (I've one that supersedes #21143 which is #24702

**Comment by vmstan** [comment]
Fixed by #24702
